defold-spine/src/script_spine.cpp

```diff
@@ -327,15 +325,50 @@ namespace dmSpine
 
         dmMessage::URL sender;
         dmScript::GetURL(L, &sender);
-        if (!CompSpineModelPlayAnimation(component, &msg, &sender, functionref, L))
+        if (!CompSpineModelPlayAnimation(component, &msg, &sender, cbk, L))
         {
+            // destroy callback immediately if error happened
+            if (cbk)
+            {
+                dmScript::DestroyCallback(cbk);
+            }
             char buffer[128];
             dmLogError("Failed to run animation '%s' on component '%s'", lua_tostring(L, 2), dmScript::UrlToString(&receiver, buffer, sizeof(buffer)));
         }
 
         return 0;
     }
 
+    void DestroyCallback(void* callback_data)
+    {
+        dmScript::LuaCallbackInfo* cbk = (dmScript::LuaCallbackInfo*)callback_data;
+        dmScript::DestroyCallback(cbk);
+    }
+
+    void RunTrackCallback(void* callback_data, const dmDDF::Descriptor* desc, const char* data, const dmMessage::URL* sender)
```

Comment:
I think it's ok for the comp_spine_model.cpp to include script.h, so that it doesn't have to pass a `void*`

Reply:
Maybe for now yes, until it's only Lua